sync/landing.py

```diff
@@ -670,7 +670,32 @@ def land_to_gecko(git_gecko, git_wpt, prev_wpt_head=None, new_wpt_head=None,
 
 @base.entry_point("landing")
 def try_push_complete(git_gecko, git_wpt, try_push, sync, allow_push=True):
-    log_files = try_push.download_raw_logs()
+    # min rate of job success to proceed with metadata update
+    target_rate = 0.7
+    retriggered = try_push.retriggered_wpt_states(force_update=True)
+    intermittents = []
+    if not try_push.success and not retriggered:
+        if try_push.success_rate < target_rate:
+            sync.error = (
+                "Latest try push for bug %s has too many failures.\n"
+                "See %s"
+            ) % (sync.bug, try_push.treeherder_url(try_push.try_rev))
+            try_push.status = "complete"
+            return
+        num_new_jobs = try_push.retrigger_failures()
+        logger.info("%s new tasks scheduled on try for %s" (num_new_jobs, sync.bug))
+        if num_new_jobs:
+            env.bz.comment(sync.bug,
+                           ("Retriggered failing web-platform-test tasks on "
+                            "try before final metadata update."))
+            return
+    for name, data in retriggered.iteritems():
+        total = sum(data["states"].itervalues()) * 1.0
```

Comment:
So this doesn't quite work. The problem is that if we have a test that reliably fails on OSX (for example) but passes on Linux then afaict we will not download the logs for the Linux run, and so when we do the metadata update it will conclude that the test fails on all platforms.

I think we need to arrange things so that we have at least one run of each test run on each platform in the logs we download.

A second problem is what about cases where we get an intermittent mixed in with some real failures. We possibly need to update the metadata update code so that for tests with multiple results we only consider the most common result.

Reply:
I'll change the exclusion to keep any logs that represent a "success".

As for the mix of intermittent and real failures, I'll follow up upstream.